### ISSUE TITLE
env.gateway doesn't use user from ssh_config

### ISSUE THREAD

**Issue by em-cliqz** [fabric/fabric#892]
It appears that the selection of user to connect to a gateway isn't always valid. We seem to have different behaviour on Linux/Mac but I think I can identify the code which may be to blame. 

In "normalize" in "network.py" the "env.user" / "env.local_user" influence which user is used to connect to the gateway. However, this means that if the ssh_config specifies a user it will not be used. This logic exists presumably so that you can connect to the remote machine as a specific user. For the gateway connection it doesn't make sense however as you would tend to only connect to the gateway as one user.

A current workaround is to specify the username in the 'env.gateway' string itself (yet for some reason this isn't required on Linux, only MacOS).

To sum up, the issue is that only the 'env.gateway' variable should override settings in the ssh_config file when connecting to the gateway. All the other host/user/port settings relate to the target host and thus should not be used here.

Possibly #884 is also a result of the same problem.
